### PR TITLE
Reinstate PaymentsLearner report

### PR DESCRIPTION
**Pull request by benagain**

### CODE REVIEW

File: src/SFA.DAS.IdentifyDataLocks.Web/Pages/PaymentsLearner.cshtml.cs [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.IdentifyDataLocks.Web.Pages
+{
+    public class PaymentsLearnerModel : PageModel
+    {
+        [BindProperty(SupportsGet = true)]
+        public string Uln { get; set; } = "9000000407";
```
Comment (DevenKShah) [inline src/SFA.DAS.IdentifyDataLocks.Web/Pages/PaymentsLearner.cshtml.cs:17]: Do we need this hard coded value?

File: src/SFA.DAS.IdentifyDataLocks.Web/Pages/PaymentsLearner.cshtml.cs [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+            CollectionPeriods = earnings.OrderBy(x => x.Key).Select(periods => new CollectionPeriod
+            {
+                PeriodName = $"R0{periods.Key}",
+                PriceEpisodes = periods.SelectMany(x => x.PriceEpisodes, (period, earning) => MapPriceEpisode(period, earning, paidCommitments, lockedCommitments, apprenticeships)).ToList(),
+            }).ToList();
+        }
+
+        private async Task<List<IGrouping<byte, EarningEventModel>>> GetEarningEvents(long learnerUln)
+        {
+            //var a = Bob(learnerUln);
```
Comment (DevenKShah) [inline src/SFA.DAS.IdentifyDataLocks.Web/Pages/PaymentsLearner.cshtml.cs:90]: Commented code can be removed?

File: src/SFA.DAS.IdentifyDataLocks.Web/Pages/PaymentsLearner.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+            var bs = await GetEarningEventsBase(learnerUln);
+            var cs = bs
+                .GroupBy(x => x.CollectionPeriod)
+                .ToList();
+
+            return cs
+                .SelectMany(x => x/*.OrderByDescending(y => y.IlrSubmissionDateTime)*/.Take(1))
```
Comment (DevenKShah) [inline src/SFA.DAS.IdentifyDataLocks.Web/Pages/PaymentsLearner.cshtml.cs:97]: Commented code

File: src/SFA.DAS.IdentifyDataLocks.Web/Pages/PaymentsLearner.cshtml.cs [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+        {
+            return context.EarningEvent
+                .Where(x => x.LearnerUln == learnerUln).ToListAsync();
+        }
+
+        private async Task<List<PaymentModel>> GetCommitments(long learnerUln)
+        {
+            var paidCommitments = await context.Payment
+                .Where(x => x.LearnerUln == learnerUln)
+                //.Where(x => x.PriceEpisodeIdentifier)
```
Comment (DevenKShah) [inline src/SFA.DAS.IdentifyDataLocks.Web/Pages/PaymentsLearner.cshtml.cs:121]: Commented code

File: src/SFA.DAS.IdentifyDataLocks.Web/Pages/PaymentsLearner.cshtml.cs [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+                ;
+            return paidCommitments;
+        }
+
+        private async Task<List<DataLockEventModel>> GetLockedCommitments(long learnerUln)
+        {
+            var lockedCommitmentsQueryable = context.DataLockEvent
+                .Include(de => de.NonPayablePeriods)
+                .ThenInclude(npp => npp.DataLockEventNonPayablePeriodFailures)
+                .Where(x => x.LearnerUln == learnerUln);//.ToList();
```
Comment (DevenKShah) [inline src/SFA.DAS.IdentifyDataLocks.Web/Pages/PaymentsLearner.cshtml.cs:132]: Commented code

File: src/SFA.DAS.IdentifyDataLocks.Web/Pages/PaymentsLearner.cshtml.cs [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+using SFA.DAS.Payments.Application.Repositories;
+using SFA.DAS.Payments.Model.Core.Audit;
+using SFA.DAS.Payments.Model.Core.Entities;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.IdentifyDataLocks.Web.Pages
+{
+    public class PaymentsLearnerModel : PageModel
```
Comment (DevenKShah) [inline src/SFA.DAS.IdentifyDataLocks.Web/Pages/PaymentsLearner.cshtml.cs:14]: The model is quiet busy, I appreciate that this is for internal support.

Comment (DevenKShah) [inline src/SFA.DAS.IdentifyDataLocks.Web/Pages/PaymentsLearner.cshtml.cs:14]: You need to add authorization for this page.

**Review by DevenKShah — CHANGES_REQUESTED** [review]
Looks good, the only change is the missing page authorization.

**Comment by benagain** [comment]
> 
> 
> Looks good, the only change is the missing page authorization.

Yes!  Thank you for spotting that.

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**